doc/rules/avoid_unused_parameters.md

```diff
@@ -0,0 +1,54 @@
+# Avoid unused parameters
+
+## Rule id
+
+avoid-unused-parameters
+
+## Description
+
+Checks for unused parameters inside a function or method body.
+For overridden methods suggests renaming unused parameters to \_, \_\_, etc.
+
+Note: abstract classes are completely ignored by the rule to avoid redundant checks for potentially overridden methods.
+
+### Example
+
+Bad:
+
+```dart
+void someFunction(String s) {
+  return;
+}
+
+class SomeClass {
+  void method(String s) {
+    return;
+  }
+}
+```
+
+Good:
+
+```dart
+void someFunction() {
```

Comment:
I guess you can add a comment here about that rule isn't triggered because there is no unused parameter, not because it was deleted as a fix from `Bad` example.

Reply:
Hmm... does it really matter? Will it be clear if i just rename function to `someOtherFunction`?

Reply:
It's just at first I read the "good" example as the "fixed" version. Renaming the function should help clear it up.